internal/buffer/line_array.go

```diff
@@ -206,6 +207,8 @@ func (la *LineArray) newlineBelow(y int) {
 
 // Inserts a byte array at a given location
 func (la *LineArray) insert(pos Loc, value []byte) {
+	la.lock.Lock()
```

Comment:
Hmm, also need locking in Buffer's `Retab()`.

Reply:
No, why? `ReplaceRegex()` just uses `insert()` and `remove()` under the hood, as far as I can see, and `FindMatchingBrace()` doesn't modify the line array at all. It's just `Retab()` that directly modifies `b.lines[i].data`. (Unless you spot any other functions that directly modify `b.lines[i].data` and/or add or remove lines from `b.lines`?).

Reply:
You're right, `FindMatchingBrace()` and `ReplaceRegex()` is just read access.